### PR TITLE
Feature/spk 89

### PR DESCRIPTION
**Pull request by AlexMisiulia**

### CODE REVIEW

File: app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt [hunk 1]
```diff
@@ -23,7 +22,16 @@ abstract class BaseNfcFragment<T: BaseNfcViewModel> : Fragment(), NfcEventListen
     private val nfcHelper: NfcHelper = Injector.appComponent.getNfcHelper()
     protected val viewModelFactory: ViewModelFactory = Injector.appComponent.getViewModelFactory()
 
+    /**
+     * This method is called to create and store ViewModel instance.
+     * No need to store this reference in your code. You can access viewModel field from this class
+     * */
     abstract fun createViewModel() : T
```
Comment (cquemin) [inline app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt]: You should also say when, in the lifecycle,  this method is called.

File: app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt [hunk 2]
```diff
@@ -23,7 +22,16 @@ abstract class BaseNfcFragment<T: BaseNfcViewModel> : Fragment(), NfcEventListen
...
+
+    /**
+     * This method is called after BaseNfcFragment and its components were initialized.
+     * You should initialize Fragment logic in this method
+     * */
     abstract fun init()
```
Comment (cquemin) [inline app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt]: Again, mentioning where in the lifecycle could be helpful to the user.

File: app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt (hunk 1 above)
Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt]: Done

File: app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt (hunk 2 above)
Comment (AlexMisiulia) [reply app/src/main/java/com/simplifire/yotinfcclient/nfc/BaseNfcFragment.kt]: Done

**Pull request merged**